packages/bus-dereference/lib/ActorDereference.ts

```diff
@@ -0,0 +1,106 @@
+import type { IAction, IActorArgs, IActorOutput, IActorTest, Mediate } from '@comunica/core';
+import { ActorDereferenceBase } from './ActorDereferenceBase';
+
+/**
+ * A base actor for dereferencing URLs to (generic) streams.
+ *
+ * Actor types:
+ * * Input:  IActionDereference:      A URL.
+ * * Test:   <none>
+ * * Output: IActorDereferenceOutput: A Readable stream
+ *
+ * @see IActionDereference
+ * @see IActorDereferenceOutput
+ */
+export abstract class ActorDereference extends
+  ActorDereferenceBase<IActionDereference, IActorTest, IActorDereferenceOutput> {
+  /**
+   * @param args - @defaultNested {<default_bus> a <cc:components/Bus.jsonld#Bus>} bus
+   */
+  public constructor(args: IActorDereferenceArgs) {
+    super(args);
+  }
+
+  /**
+   * Handle the given error as a rejection or delegate it to the logger,
+   * depending on whether or not hard errors are enabled.
+   * @param {IActionDereference} action A dereference action.
+   * @param {Error} error An error that has occurred.
+   * @param headers Optional HTTP headers to pass.
+   * @param {number} requestTime The time it took to request the page in milliseconds.
+   * @return {Promise<IActorDereferenceOutput>} A promise that rejects or resolves to an empty output.
+   */
+  protected async handleDereferenceErrors(
+    action: IActionDereference,
+    error: unknown,
+    headers?: Headers | undefined,
+    requestTime = 0,
+  ): Promise<IActorDereferenceOutput> {
+    return this.dereferenceErrorHandler(action, error, { url: action.url, exists: false, headers, requestTime });
+  }
+}
+
+export interface IActionDereference extends IAction {
+  /**
+   * The URL to dereference
+   */
+  url: string;
+  /**
+   * By default, actors will reject upon receiving non-200 HTTP responses.
+   * If this option is true, then all HTTP responses will cause the action to resolve,
+   * but some outputs may therefore contain empty quad streams.
+   */
+  acceptErrors?: boolean;
+  /**
+   * Optional HTTP method to use.
+   * Defaults to GET.
+   */
+  method?: string;
+  /**
+   * Optional HTTP headers to pass.
+   */
+  headers?: Headers;
+  /**
+   * An optional callback to retrieve the mediaType mappings
+   */
+  mediaTypes?: () => Promise<Record<string, number> | undefined>;
+}
+
+interface IReadableClose extends NodeJS.ReadableStream {
```

Comment:
Is there a better type I can use here?

Reply:
Maybe just `AsyncIterator`? Comunica already has a strong dependency on it anyways. But the way you solved it now also seems ok to me.

Reply:
Lets just leave it as is then :)